api-flask/app/nginx.conf

```diff
@@ -0,0 +1,59 @@
+# Nginx config.
+
+# Allow large URI.
+client_header_buffer_size 64k;
+large_client_header_buffers 4 64k;
+
+# Avoid timeouts for large files.
+proxy_connect_timeout       2000;
+proxy_send_timeout          2000;
+proxy_read_timeout          2000;
+send_timeout                2000;
+
+uwsgi_read_timeout          2000;
+uwsgi_send_timeout          2000;
+
+# Avoid 413s for large POST bodies.
+client_max_body_size        30m;
+
+# Http server settings.
+server {
+
+    listen 80;
+
+    add_header "Access-Control-Allow-Origin" "*";
+
+    location / {
+        if ($request_method = "OPTIONS") {
+          
+          add_header "Access-Control-Allow-Methods" "GET, POST, OPTIONS";
+          #
+          # Custom headers and headers various browsers *should* be OK with but aren"t
+          #
+          add_header "Access-Control-Allow-Headers" "DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range";
+          #
+          # Tell client that this pre-flight info is valid for 20 days
+          #
+          add_header "Access-Control-Max-Age" 1728000;
+          add_header "Content-Type" "text/plain; charset=utf-8";
+          add_header "Content-Length" 0;
+          return 204;
+       }
```

Comment:
Isn't the point of adding a CORS configuration in the Flask app to let it handle the preflight requests? I can see why it might be nice to have the proxy do it without involving the app, but it seems like this route only applies to `/`, not `/stats`, which is where the actual requests are going to get sent. Do we need this definition here at all?